contrib/babelfishpg_tsql/src/pltsql_utils.c

```diff
@@ -3093,3 +3093,146 @@ is_valid_func_ownership_chain(void *expr, Oid objectOwnerId)
 	}
 	return (OidIsValid(immediate_parent_func) && (get_func_owner(immediate_parent_func) == objectOwnerId));
 }
+
+/*
+ * restrict_alter_owner_stmt
+ * Blocks ALTER OWNER statements from PG endpoint on TSQL objects.
+ * Allows ALTER OWNER for specific shipped objects only when new owner is sysadmin.
+ */
+void
+restrict_alter_owner_stmt(AlterOwnerStmt *stmt)
+{
+    Oid			schema_oid = InvalidOid;
+    char		*schema_name = NULL;
+    char		*object_name = NULL;
+    ObjectAddress	address;
+    Relation		relation = NULL;
+    Node			*object = stmt->object;
+
+    /* Only handle specific object types */
+    if (stmt->objectType != OBJECT_TYPE && stmt->objectType != OBJECT_SCHEMA &&
+        stmt->objectType != OBJECT_FUNCTION && stmt->objectType != OBJECT_PROCEDURE)
```

Comment:
We create a lot of other types of objects in babelfish, what about them? Like OBJECT_DOMAIN, OBJECT_OPERATOR, OBJECT_ROUTINE, OBJECT_COLLATION etc. Why do we want to restrict to only these?

Reply:
Those are not user defined objects. We restrict only these objects mentioned in the [doc](https://quip-amazon.com/deUEAW2xBdaH/Altering-the-owner-of-Babelfish-Objects-should-be-blocked-via-PG-endpoint).

Reply:
We might need to revisit this.